src/CashuWallet.ts

```diff
@@ -194,96 +331,286 @@ class CashuWallet {
 		return proofs;
 	}
 
+	/**
+	 * Send proofs of a given amount, by providing at least the required amount of proofs
+	 * @param amount amount to send
+	 * @param proofs array of proofs (accumulated amount of proofs must be >= than amount)
+	 * @param options.outputAmounts? optionally specify the output's amounts to keep and send.
+	 * @param options.counter? optionally set counter to derive secret deterministically. CashuWallet class must be initialized with seed phrase to take effect
+	 * @param options.proofsWeHave? optionally provide all currently stored proofs of this mint. Cashu-ts will use them to derive the optimal output amounts
+	 * @param options.pubkey? optionally locks ecash to pubkey. Will not be deterministic, even if counter is set!
+	 * @param options.privkey? will create a signature on the output secrets if set
+	 * @param options.keysetId? override the keysetId derived from the current mintKeys with a custom one. This should be a keyset that was fetched from the `/keysets` endpoint
+	 * @param options.offline? optionally send proofs offline.
+	 * @param options.includeFees? optionally include fees in the response.
+	 * @returns {SendResponse}
+	 */
+	async send(
+		amount: number,
+		proofs: Array<Proof>,
+		options?: {
+			outputAmounts?: OutputAmounts;
+			proofsWeHave?: Array<Proof>;
+			counter?: number;
+			pubkey?: string;
+			privkey?: string;
+			keysetId?: string;
+			offline?: boolean;
+			includeFees?: boolean;
+		}
+	): Promise<SendResponse> {
+		if (sumProofs(proofs) < amount) {
+			throw new Error('Not enough funds available to send');
+		}
+		const { keep: keepProofsOffline, send: sendProofOffline } = this.selectProofsToSend(
```

Comment:
doesn't this need to consider `outputAmounts`, in case `includeFees` is set to `true`? Or how do we make sure the fees are covered? `IncludeFees` is thought to cover the fees for the receiver, right?

Reply:
we also should add a function, to return the fees to the user, before we perform the split. I think most wallets will need to show that:

1. user enters amount
2. display amount + fees = total leaving the wallet
3. perform send

Reply:
> doesn't this need to consider `outputAmounts`, in case `includeFees` is set to `true`? Or how do we make sure the fees are covered? `IncludeFees` is thought to cover the fees for the receiver, right?

I don't understand. `includeFees` should includes the fees in `sendProofOffline`. What does this have to do with outputs?

Reply:
I think this can be resolved. `outputAmounts` does only determine changeOutputs kept by the sender. To cover the fees for the receiving end we either include then in the sent proofs in `selectProofsToSend` or in `swap`. If `offline` is true, fee is included in `sendProofOffline` and if we swap it is included in `send` returned by `swap`

[ab2fe7e5-d4f9-47f8-9a81-1ef6b0ec332a](https://github.com/cashubtc/cashu-ts/pull/159/files#diff-ed2a551c11fe95a44ccc731d64f73eea7ced34250da86272f7ec5a0ad6a8781bR371)